packages/ui/src/components/va-tree-view/types.ts

```diff
@@ -10,6 +10,7 @@ export interface TreeNode {
   hasChildren?: boolean
   matchesFilter?: boolean
   indeterminate?: boolean
+  children: TreeNode[]
```

Comment:
Shouldn't be required, see https://github.com/epicmaxco/vuestic-ui/issues/3077.

Reply:
Hmm, okay. I made it optional. But now I think that we probably need different 'Types' for nodes. The first type is for user data, 2nd type is for our internal node structure. `buildTree` method always returns a node with the `children` property, but we definitely shouldn't force users to put empty arrays.

Reply:
User tree node item I would assume should be generic on item type https://blog.vuejs.org/posts/vue-3-3#generic-components.